File: Src/Icm.Core.Tests/Basic types extensions/StringExtensionsTest.vb
Imports Icm

'''<summary>
'''This is a test class for StringExtensions and is intended
'''to contain all StringExtensions Unit Tests
'''</summary>
<TestFixture(), Category("Icm")>
Public Class StringExtensionsTest

    <TestCase("bLioNNpPA", "BLioNNpPA")>
    <TestCase("BaMBu", "BaMBu")>
    <TestCase("áNGeL", "ÁNGeL")>
    <TestCase("78a", "78a")>
    <TestCase("a1", "A1")>
    <TestCase("", "")>
    <TestCase(Nothing, "")>
    Public Sub ToUpperFirstTest_BasicCase(s As String, expected As String)
        Dim actual As String
        actual = s.ToUpperFirst()
        Assert.AreEqual(expected, actual)
    End Sub

    <TestCase("aa", 3, "aaaaaa", Nothing)>
    <TestCase("", 3, "", Nothing)>
    <TestCase("aa", 0, "", Nothing)>
    <TestCase("aa", -5, Nothing, GetType(ArgumentOutOfRangeException))>
    <TestCase(Nothing, 3, Nothing, GetType(NullReferenceException))>
    Public Sub RepeatTest(s As String, count As Integer, expected As String, expectedException As Type)
        Dim actual As String

        If expectedException Is Nothing Then
            actual = s.Repeat(count)
            Assert.AreEqual(expected, actual)
        Else
            Assert.That(Sub() s.Repeat(count), Throws.TypeOf(expectedException))
        End If
    End Sub


    '''<summary>
    '''A test for Left
    '''</summary>
    <Test()>
    Public Sub LeftTest()
        Dim s As String = String.Empty
        Dim length As Integer = 0
        Dim expected As String = String.Empty
        Dim actual As String

        'Caso 1
        s = "Maria"
        length = 3
        expected = "Mar"
        actual = StringExtensions.Left(s, length)
        Assert.AreEqual(expected, actual)

        'Caso 2
        s = "M12a"
        length = 3
        expected = "M12"
        actual = StringExtensions.Left(s, length)
        Assert.AreEqual(expected, actual)

        'Caso 3
        s = "Viernes"
        length = 12
        expected = ""
        Try
            actual = StringExtensions.Left(s, length)
            Assert.Fail("Should throw ArgumentOutOfRangeException")
        Catch ex As ArgumentOutOfRangeException

        Catch ex As Exception
            Assert.Fail("Should throw ArgumentOutOfRangeException")
        End Try

        'Caso 4
        s = "Maria"
        length = -2
        expected = "Mar"
        Try
            actual = StringExtensions.Left(s, length)
            Assert.Fail("Should throw ArgumentOutOfRangeException")
        Catch ex As ArgumentOutOfRangeException

        Catch ex As Exception
            Assert.Fail("Should throw ArgumentOutOfRangeException")
        End Try

    End Sub

    '''<summary>
    '''A test for Med
    '''</summary>
    <Test()>
    Public Sub MedTest()
        Dim s As String = String.Empty
        Dim startIdx As Integer = 0
        Dim endIdx As Integer = 0
        Dim expected As String = String.Empty
        Dim actual As String

        'Caso 1
        s = "ViernesLunes"
        startIdx = 2
        endIdx = 8
        expected = "ernesLu"
        actual = StringExtensions.Med(s, startIdx, endIdx)
        Assert.AreEqual(expected, actual)

        'Caso 2
        s = "ViernesLunes"
        startIdx = 0
        endIdx = 0
        expected = "V"
        actual = StringExtensions.Med(s, startIdx, endIdx)
        Assert.AreEqual(expected, actual)

        'Caso 3
        s = "ViernesLunes"
        startIdx = -4
        endIdx = -6

        Try
            actual = StringExtensions.Med(s, startIdx, endIdx)
            Assert.Fail("Should throw ArgumentOutOfRangeException")
        Catch ex As ArgumentOutOfRangeException

        Catch ex As Exception
            Assert.Fail("Should throw ArgumentOutOfRangeException")
        End Try



    End Sub

    '''<summary>
    '''A test for SkipBoth
    '''</summary>
    <Test()>
    Public Sub SkipBothTest()
        Dim s As String = String.Empty '
        Dim startLength As Integer = 0 ' 
        Dim endLength As Integer = 0 '
        Dim expected As String = String.Empty
        Dim actual As String

        'Caso 1
        s = "LunesMartes"
        startLength = 1
        endLength = 3
        expected = "unesMar"
        actual = StringExtensions.SkipBoth(s, startLength, endLength)
        Assert.AreEqual(expected, actual)

        'Caso 2
        s = "LunesMartes"
        startLength = -1
        endLength = -5
        expected = ""

        Try
            actual = StringExtensions.SkipBoth(s, startLength, endLength)
            Assert.Fail("Should throw ArgumentOutOfRangeException")
        Catch ex As ArgumentOutOfRangeException

        Catch ex As Exception
            Assert.Fail("Should throw ArgumentOutOfRangeException")
        End Try

        'Caso 3
        s = "LunesMartes"
        startLength = 0
        endLength = 0
        expected = "LunesMartes"
        actual = StringExtensions.SkipBoth(s, startLength, endLength)
        Assert.AreEqual(expected, actual)

    End Sub

    '''<summary>
    '''A test for Right
    '''</summary>
    <Test()>
    Public Sub RightTest()
        Dim s As String = String.Empty
        Dim length As Integer = 0
        Dim expected As String = String.Empty
        Dim actual As String

        'Caso 1
        s = "Maria"
        length = 3
        expected = "ria"
        actual = StringExtensions.Right(s, length)
        Assert.AreEqual(expected, actual)

        'Caso 2
        s = "M12a"
        length = 3
        expected = "12a"
        actual = StringExtensions.Right(s, length)
        Assert.AreEqual(expected, actual)

        'Caso 3
        s = "Viernes"
        length = 12
        expected = ""
        Try
            actual = StringExtensions.Right(s, length)
            Assert.Fail("Should throw ArgumentOutOfRangeException")
        Catch ex As ArgumentOutOfRangeException

        Catch ex As Exception
            Assert.Fail("Should throw ArgumentOutOfRangeException")
        End Try

        'Caso 4
        s = "Maria"
        length = -2
        expected = "Mar"
        Try
            actual = StringExtensions.Right(s, length)
            Assert.Fail("Should throw ArgumentOutOfRangeException")
        Catch ex As ArgumentOutOfRangeException

        Catch ex As Exception
            Assert.Fail("Should throw ArgumentOutOfRangeException")
        End Try
    End Sub

    '''<summary>
    '''A test for SurroundedBy
    '''</summary>
    <Test()>
    Public Sub SurroundedByTest()
        Dim s As String = String.Empty '
        Dim startS As String = String.Empty
        Dim endS As String = String.Empty
        Dim expected As Boolean = False ' 
        Dim actual As Boolean

        'Caso1
        s = "LaCasaBonita"
        startS = "La"
        endS = "ta"
        expected = True

        actual = StringExtensions.SurroundedBy(s, startS, endS)
        Assert.AreEqual(expected, actual)


        'Caso 2
        s = "ángela"
        startS = "án"
        endS = "la"
        expected = True

        actual = StringExtensions.SurroundedBy(s, startS, endS)
        Assert.AreEqual(expected, actual)

        'Caso3
        s = "Concierto"
        startS = "conc"
        endS = "a"
        expected = False

        actual = StringExtensions.SurroundedBy(s, startS, endS)
        Assert.AreEqual(expected, actual)



    End Sub
End Class

File: Src/Icm.Core.Tests/Basic types extensions/StringExtensionsTest.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using Icm;
using NUnit.Framework;

[TestFixture(), Category("Icm")]
public class StringExtensionsTest
{

	[TestCase("bLioNNpPA", ExpectedResult = "BLioNNpPA")]
	[TestCase("BaMBu", ExpectedResult = "BaMBu")]
	[TestCase("áNGeL", ExpectedResult = "ÁNGeL")]
	[TestCase("78a", ExpectedResult = "78a")]
	[TestCase("a1", ExpectedResult = "A1")]
	[TestCase("", ExpectedResult = "")]
	[TestCase(null, ExpectedResult = "")]
	public string ToUpperFirst_Test(string s)
	{
		return s.ToUpperFirst();
	}

	[TestCase("aa", 3, ExpectedResult = "aaaaaa")]
	[TestCase("", 3, ExpectedResult = "")]
	[TestCase("aa", 0, ExpectedResult = "")]
	[TestCase("aa", -5, ExpectedException = typeof(ArgumentOutOfRangeException))]
	[TestCase(null, 3, ExpectedException = typeof(NullReferenceException))]
	public string Repeat_Test(string s, int count)
	{
		return s.Repeat(count);
	}


	[TestCase("Maria", 3, ExpectedResult = "Mar")]
	[TestCase("M12a", 3, ExpectedResult = "M12")]
	[TestCase("Maria", 0, ExpectedResult = "")]
	[TestCase("Maria", 12, ExpectedException = typeof(ArgumentOutOfRangeException))]
	[TestCase("Maria", -12, ExpectedException = typeof(ArgumentOutOfRangeException))]
	public string Left_Test(string target, int length)
	{
		return target.Left(length);
	}

	[TestCase("ViernesLunes", 2, 8, ExpectedResult = "ernesLu")]
	[TestCase("ViernesLunes", 0, 0, ExpectedResult = "V")]
	[TestCase("ViernesLunes", 5, 4, ExpectedResult = "")]
	[TestCase("ViernesLunes", 0, -1, ExpectedResult = "")]
	[TestCase("ViernesLunes", 5, 3, ExpectedException = typeof(ArgumentOutOfRangeException))]
	[TestCase("ViernesLunes", -4, 6, ExpectedException = typeof(ArgumentOutOfRangeException))]
	[TestCase("ViernesLunes", 4, -6, ExpectedException = typeof(ArgumentOutOfRangeException))]
	[TestCase("ViernesLunes", 20, 21, ExpectedException = typeof(ArgumentOutOfRangeException))]
	public string Med_Test(string target, int startIdx, int endIdx)
	{
		return target.Med(startIdx, endIdx);
	}

	[TestCase("LunesMartes", 1, 3, ExpectedResult = "unesMar")]
	[TestCase("LunesMartes", 0, 0, ExpectedResult = "LunesMartes")]
	[TestCase("LunesMartes", 5, 6, ExpectedResult = "")]
	[TestCase("LunesMartes", 5, 7, ExpectedException = typeof(ArgumentOutOfRangeException), Description = "Overlapped lengths")]
	[TestCase("LunesMartes", -1, 3, ExpectedException = typeof(ArgumentOutOfRangeException))]
	[TestCase("LunesMartes", 1, -3, ExpectedException = typeof(ArgumentOutOfRangeException))]
	[TestCase("LunesMartes", 20, 3, ExpectedException = typeof(ArgumentOutOfRangeException))]
	[TestCase("LunesMartes", 2, 23, ExpectedException = typeof(ArgumentOutOfRangeException))]
	public string SkipBoth_Test(string target, int startLength, int endLength)
	{
		return target.SkipBoth(startLength, endLength);
	}

	[TestCase("Maria", 3, ExpectedResult = "ria")]
	[TestCase("M12a", 3, ExpectedResult = "12a")]
	[TestCase("Maria", 0, ExpectedResult = "")]
	[TestCase("Maria", 12, ExpectedException = typeof(ArgumentOutOfRangeException))]
	[TestCase("Maria", -12, ExpectedException = typeof(ArgumentOutOfRangeException))]
	public string Right_Test(string target, int length)
	{
		return target.Right(length);
	}

	///<summary>
	///A test for SurroundedBy
	///</summary>
	[TestCase("LaCasaBonita", "La", "Bonita", ExpectedResult = true)]
	[TestCase("LaCasaBonita", "Li", "Bonita", ExpectedResult = false)]
	[TestCase("LaCasaBonita", "La", "Fea", ExpectedResult = false)]
	public bool SurroundedBy_Test(string target, string startString, string endString)
	{
		return target.SurroundedBy(startString, endString);
	}

	[TestCase("qwer", "subst", ExpectedResult = "qwer")]
	[TestCase("", "subst", ExpectedResult = "subst")]
	[TestCase(null, "subst", ExpectedResult = null)]
	public string IfEmpty_Test(string target, string ifEmptyString)
	{
		return target.IfEmpty(ifEmptyString);
	}

	[TestCase("qwer", ExpectedResult = false)]
	[TestCase("", ExpectedResult = true)]
	[TestCase(null, ExpectedResult = false)]
	public bool IsEmpty_Test(string target)
	{
		return target.IsEmpty();
	}
}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
